core/primitives/transaction_validity.hpp

```diff
@@ -84,6 +84,77 @@ namespace kagome::primitives {
    */
   using TransactionValidity = boost::variant<Invalid, Valid, Unknown>;
 
+  /**
+   * @brief outputs object of type Invalid to stream
+   * @tparam Stream stream type
+   * @param s stream reference
+   * @param v value to output
+   * @return reference to stream
+   */
+  template <class Stream>
+  Stream &operator<<(Stream &s, const Invalid &v) {
+    return s << v.error_;
+  }
+
+  /**
+   * @brief outputs object of type Valid to stream
+   * @tparam Stream stream type
+   * @param s stream reference
+   * @param v value to output
+   * @return reference to stream
+   */
+  template <class Stream>
+  Stream &operator<<(Stream &s, const Valid &v) {
+    return s << v.priority_ << v.requires_ << v.provides_ << v.longevity_;
+  }
+
+  /**
+   * @brief outputs object of type Unknown to stream
+   * @tparam Stream stream type
+   * @param s stream reference
+   * @param v value to output
+   * @return reference to stream
+   */
+  template <class Stream>
+  Stream &operator<<(Stream &s, const Unknown &v) {
+    return s << v.error_;
+  }
+
+  /**
+   * @brief decodes object of type Invalid from stream
+   * @tparam Stream input stream type
+   * @param s stream reference
+   * @param v value to decode
+   * @return reference to stream
+   */
+  template <class Stream>
+  Stream &operator>>(Stream &s, Invalid &v) {
+    return s >> v.error_;
+  }
+
+  /**
+   * @brief decodes object of type Valid from stream
+   * @tparam Stream input stream type
+   * @param s stream reference
+   * @param v value to decode
+   * @return reference to stream
+   */
+  template <class Stream>
+  Stream &operator>>(Stream &s, Valid &v) {
+    return s >> v.priority_ >> v.requires_ >> v.provides_ >> v.longevity_;
```

Comment:
also, please remove all suffixes `_` from all public members in structs

Reply:
done